editor/components/block-list/invalid-block-warning.js

```diff
@@ -20,46 +20,32 @@ import {
 import { replaceBlock } from '../../store/actions';
 import Warning from '../warning';
 
-function InvalidBlockWarning( { ignoreInvalid, switchToBlockType } ) {
-	const htmlBlockName = 'core/html';
-	const defaultBlockType = getBlockType( getUnknownTypeHandlerName() );
-	const htmlBlockType = getBlockType( htmlBlockName );
-	const switchTo = ( blockType ) => () => switchToBlockType( blockType );
+function InvalidBlockWarning( { block, onReplace } ) {
+	const hasHTMLBlock = !! getBlockType( 'core/html' );
+
+	const convertToHTML = () => {
```

Comment:
Could these be passed in as props via `connect`s `mapDispatchToProps`? I see it as an improvement in that:

- This is a presentational component, and the logic here is not very presentational
- `mapDispatchToProps` should only create a single function reference for the lifetime of the component, avoiding unnecessary re-renders (`function() {} !== function() {}`)

Reply:
Sounds good.

Reply:
Done.